### ISSUE TITLE
Enable block cursor

### ISSUE THREAD

**Issue by osa1** [equalsraf/neovim-qt#202]
As far as I understand neovim-qt ignores `guicursor` settings and uses a bar cursor in insert mode. It'd be great if existing `guicursor` settings worked, or alternatively, if `neovim-qt` provided its own settings for using block cursor.

**Comment by osa1** [comment]
Any hope for this? Ping @equalsraf 

I'm not happy with my terminal's (st) rendering speed. neovim-qt is much better, but the line cursor is making it unusable for me.

**Comment by osa1** [comment]
```patch
diff --git a/src/gui/shell.cpp b/src/gui/shell.cpp
index 526ec5c..498fdd1 100644
--- a/src/gui/shell.cpp
+++ b/src/gui/shell.cpp
@@ -593,10 +593,7 @@ void Shell::paintEvent(QPaintEvent *ev)
 		bool wide = contents().constValue(m_cursor_pos.y(),
 						m_cursor_pos.x()).doubleWidth;
 		QRect cursorRect(neovimCursorTopLeft(), cellSize());
-
-		if (m_insertMode) {
-			cursorRect.setWidth(2);
-		} else if (wide) {
+        if (!m_insertMode && wide) {
 			cursorRect.setWidth(cursorRect.width()*2);
 		}
 		QPainter painter(this);
```

This fixes it for me.

**Comment by justinmk** [comment]
Note that nvim 0.2 has mode_info_set UI event so guicursor option can be supported by UIs.

**Comment by jgehrig** [comment]
The `guicursor` option is now supported as of Pull #520. It is available as of `v0.2.16` and the only cursor in `master`.

![blinking curosr](https://user-images.githubusercontent.com/11207308/67357314-e6bba700-f52a-11e9-921c-e62eb24bb724.gif)

Marking as closed.
